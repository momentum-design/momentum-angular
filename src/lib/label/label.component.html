
<label
  [ngClass]="labelClasses"
>
  <span>{{label}}</span>
</label>
