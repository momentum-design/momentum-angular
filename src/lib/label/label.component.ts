import { Component, OnInit, Input } from '@angular/core';

@Component({
  selector: 'cui-label',
  templateUrl: './label.component.html',
  styles: []
})
export class LabelComponent implements OnInit {

  @Input() public className: string;
  @Input() public theme: string;
  @Input() public label: string;
  @Input() public htmlFor: string;

  constructor() { }

  ngOnInit() {
  }

  get labelClasses() {
    return {
      ['cui-label--' + this.theme]: this.theme,
      [this.className]: this.className,
    };
  }

}
