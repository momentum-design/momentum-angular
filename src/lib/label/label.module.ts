import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LabelComponent } from './label.component';

@NgModule({
  declarations: [LabelComponent],
  imports: [CommonModule],
  exports: [LabelComponent],
  schemas: [CUSTOM_ELEMENTS_SCHEMA],
})
export class LabelModule {}
