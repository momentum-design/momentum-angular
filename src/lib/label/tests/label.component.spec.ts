import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { LabelComponent } from '../label.component';

describe('LabelComponent', () => {
  let component: LabelComponent;
  let fixture: ComponentFixture<LabelComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [LabelComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(LabelComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render the label text', () => {
    component.label = 'Test Label';
    fixture.detectChanges();

    const inputNativeElement = fixture.nativeElement;
    const span = inputNativeElement.querySelector('label > span');

    expect(span).not.toBeNull();
    expect(span.textContent).toMatch('Test Label');
  });

  it('should render a theme to the label', () => {
    component.theme = 'dark';
    fixture.detectChanges();

    const inputNativeElement = fixture.nativeElement;
    const label = inputNativeElement.querySelector('label');

    expect(label.className).toContain('cui-label--dark');
  });

  it('should render a custom class to the label', () => {
    component.className = 'test';
    fixture.detectChanges();

    const inputNativeElement = fixture.nativeElement;
    const label = inputNativeElement.querySelector('label');

    expect(label.className).toContain('test');
  });
});
