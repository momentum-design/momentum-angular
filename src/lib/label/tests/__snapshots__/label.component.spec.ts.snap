// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelComponent should create and match snapshot 1`] = `
<md-label>
  <label
    ng-reflect-ng-class="[object Object]"
  >
    <span>
      
    </span>
  </label>
</md-label>
`;
